aws/table_aws_s3_bucket.go

```diff
@@ -453,6 +464,46 @@ func getBucketIsPublic(ctx context.Context, d *plugin.QueryData, h *plugin.Hydra
 	return policyStatus, nil
 }
 
+func getBucketIntelligentTieringConfiguration(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Bucket location will be nil if getBucketLocation returned an error but
+	// was ignored through ignore_error_codes config arg
+	if h.HydrateResults["getBucketLocation"] == nil {
+		return nil, nil
+	}
+
+	bucket := h.Item.(types.Bucket)
+	location := h.HydrateResults["getBucketLocation"].(*s3.GetBucketLocationOutput)
+
+	// Create client
+	svc, err := S3Client(ctx, d, string(location.LocationConstraint))
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_s3_bucket.getBucketIntelligentTieringConfiguration", "client_error", err)
+		return nil, err
+	}
+
+	params := &s3.ListBucketIntelligentTieringConfigurationsInput{
+		Bucket: bucket.Name,
+	}
+
+	pageLeft := true
+	var configurations []types.IntelligentTieringConfiguration
```

Comment:
@ParthaI if this is a paginated response, shouldn't we be adding a separate table altogether?

Reply:
@misraved, It can be a separate table as pagination is supported, but the we can not pass the maximum number of result per page, as input param. Will it be worth to add it as a separate table?